Framework/DataHandling/src/LoadILLDiffraction.cpp

```diff
@@ -192,8 +200,136 @@ void LoadILLDiffraction::loadMetaData() {
+  const auto instrumentWorkspace = loadEmptyInstrument();
+  const auto &instrument = instrumentWorkspace->getInstrument();
+
+  auto scanningWorkspaceBuilder =
+      DataObjects::ScanningWorkspaceBuilder(instrument, nTimeIndexes, nBins);
+
+  std::vector<double> timeDurations =
+      getScannedVaribleByPropertyName(scan, "Time");
+  scanningWorkspaceBuilder.setTimeRanges(m_startTime, timeDurations);
+
+  g_log.debug() << "First time index starts at:"
+                << m_startTime.toISO8601String() << "\n";
+
+  g_log.debug() << "Last time index ends at:"
+                << (m_startTime + std::accumulate(timeDurations.begin(),
+                                                  timeDurations.end(), 0.0))
+                       .toISO8601String() << "\n";
+
+  // Angles in the NeXus files are the absolute position for tube 1
+  std::vector<double> tubeAngles =
+      getScannedVaribleByPropertyName(scan, "Position");
+
+  const auto &tube1Position =
+      instrument->getComponentByName("tube_1")->getPos();
+
+  // Convert the tube positions to relative rotations for all detectors
+  calculateRelativeRotations(tubeAngles, tube1Position);
+
+  auto rotationCentre = V3D(0, 0, 0);
+  auto rotationAxis = V3D(0, 1, 0);
+  scanningWorkspaceBuilder.setRelativeRotationsForScans(
+      std::move(tubeAngles), rotationCentre, rotationAxis);
+
+  m_outWorkspace = scanningWorkspaceBuilder.buildWorkspace();
+}
+
+/**
+ * Convert from absolute rotation angle, around the sample, of tube 1, to a
+ *relative rotation angle around the sample.
+ *
+ * @param tubeRotations Input is the absolute rotations around the sample of
+ *tube 1, output is the relative rotations required from the IDF for all
+ *detectors
+ * @param firstTubePosition A V3D object containing the position of the first
+ *tube
+ */
+void LoadILLDiffraction::calculateRelativeRotations(
+    std::vector<double> &tubeRotations, const V3D &firstTubePosition) {
+  // The rotations in the NeXus file are the absolute rotation of the first
+  // tube. Here we get the angle of that tube as defined in the IDF.
+
+  const double firstTubeRotationAngle =
+      firstTubePosition.angle(V3D(0, 0, 1)) * rad2deg;
+  g_log.debug() << "First tube rotation:" << firstTubeRotationAngle << "\n";
+
+  // Now pass calculate the rotations to apply for each time index.
+  std::transform(
+      tubeRotations.begin(), tubeRotations.end(), tubeRotations.begin(),
+      [&](double angle) { return (angle - firstTubeRotationAngle); });
+
+  g_log.debug() << "Instrument rotations to be applied : "
+                << tubeRotations.front() << " to " << tubeRotations.back()
+                << "\n";
+}
+
+/**
+ * Fills the counts for the instrument with moving detectors.
+ *
+ * @param data : detector data
+ * @param scan : scan data
+ */
+void LoadILLDiffraction::fillMovingInstrumentScan(const NXUInt &data,
+                                                  const NXDouble &scan) {
+
+  std::vector<double> axis = {-0.5, 0.5};
```

Comment:
This was a limitation of the `ScanningWorkspaceBuilder`, I have added an an option to use it with point data.